var/spack/repos/builtin/packages/py-jsonschema/package.py

```diff
@@ -28,3 +31,6 @@ class PyJsonschema(PythonPackage):
     depends_on('py-attrs@17.4.0:', when='@3:', type=('build', 'run'))
     depends_on('py-pyrsistent@0.14.0:', when='@3:', type=('build', 'run'))
     depends_on('py-six@1.11.0:', when='@3:', type=('build', 'run'))
+
+    depends_on('py-importlib-metadata', when='@3.1.1:', type=('build', 'run'))
+    depends_on('py-importlib-metadata', when='^python@:3.7.99', type=('build', 'run'))
```

Comment:
I think you meant to combine these:
```suggestion
    depends_on('py-importlib-metadata', when='@3.1.1: ^python@:3.7.99', type=('build', 'run'))
```